URBNSwiftCarousel/Classes/URBNSwCarouselTransitionController.swift

```diff
@@ -7,4 +7,341 @@
 //
 
 
+/**
+ Animation controller built to handle transitions between two images in separate view controllers.
+ By default, this class supports non-interactive transitions. To enable interactive transitions, register
+ a view for them using the registration method provided.
+ 
+ This class implements the UIViewControllerTransitioningDelegate protocol. As a consumer, you only need to implement
+ the URBNCarouselTransitioning protocol.
+ 
+ For the time being, this controller only supports transitions between images.
+ */
 
+
+public class URBNSwCarouselTransitionController: NSObject, UIViewControllerAnimatedTransitioning, UIViewControllerTransitioningDelegate, UIViewControllerInteractiveTransitioning, UIGestureRecognizerDelegate {
+    private(set) var interactive = false
+    private var viewInteractionBlocks = [UIView: URBNCarouselViewInteractionBeganClosure]()
```

Comment:
I'm not crazy about keeping an array of blocks like this. it might be nicer to have to views conform to a protocol which requires them to have a method for interaction. then the carousel calls that method rather than looking up a block in a dictionary